radixdlt-core/radixdlt/src/main/java/com/radixdlt/RadixNodeModule.java

```diff
@@ -192,6 +192,116 @@ protected void configure() {
 		install(new TCPTransportModule(properties));
 		install(new AddressBookModule());
 		install(new HostIpModule(properties));
+
+		// API
+		configureApi(universeType(universeModule));
+	}
+
+	private UniverseType universeType(UniverseModule universeModule) {
+		try {
+			return universeModule.universe(properties, DefaultSerialization.getInstance()).type();
```

Comment:
We shouldn't have getters in Modules as this is the purpose of dependency injection.

Reply:
This is a workaround for the situation when I need to make configuration decision using already configured in "Guice map" information. Unfortunately, this information is not easily available via other means.

Perhaps we should consider configuring Guice in "layered" fashion, when initially we configure "startup" injector with core things (like runtime properties and universe) and then assemble a working injector using "startup" injector to inject core dependencies.